mern-ecommerce/controllers/order.js

```diff
@@ -61,3 +61,15 @@ exports.updateOrderStatus = (req, res) => {
     }
   );
 };
+
+exports.purgeOrder = async (req, res) => {
+  try {
+    const result = await Order.findOneAndRemove({ _id: req.params.orderId });
+    if (!result) {
```

Comment:
In the `purgeOrder` function, the `result` variable is not checked for null before accessing its properties. It should be checked for null or undefined before accessing its properties to avoid potential null pointer exceptions.